bash_spec.bnf

```diff
@@ -0,0 +1,214 @@
+# vscode has an extension that does the highlighting for this!
+# source : https://flylib.com/books/en/4.108.1.100/1/
+
+#  D.2 BNF for bash
+
+# The following is the syntax of bash 2.0 in Backus-Naur Form (BNF):
+
+<letter> ::= a|b|c|d|e|f|g|h|i|j|k|l|m|n|o|p|q|r|s|t|u|v|w|x|y|z|
+             A|B|C|D|E|F|G|H|I|J|K|L|M|N|O|P|Q|R|S|T|U|V|W|X|Y|Z
+
+<digit> ::= 0|1|2|3|4|5|6|7|8|9
+
+<number> ::= <digit>
+           | <number> <digit>
+
+# usefull to know finally what a word actually is in shell
+# can use these rules to do the splitting of our string after tokenizing
+# maybe it's better to do this during the tokenizing?
+<word> ::= <letter>
```

Comment:
I'm confused by the fact that characters like `~`, `#`, `.`, `/`, `*` are not mentioned while definitely valid.
I feel like a word is much more than a combination of letters and underscores. For example, a path! Like, `/usr/bin` this must be a valid WORD, right?